src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -119,6 +137,20 @@ protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, C
 
                 return node;
             }
+
+            SyntaxNode? TryFindAccessibleLabel(SyntaxNode node)
+            {
+                var statement = node.FirstAncestorOrSelf<GotoStatementSyntax>();
+                if (statement is null)
+                    return null;
+
+                if (semanticModel.GetOperation(statement) is not IBranchOperation gotoOperation)
+                    return null;
+
+                Debug.Assert(gotoOperation is { BranchKind: BranchKind.GoTo });
+                var target = gotoOperation.Target;
+                return target.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
```

Comment:
i don't understand this.  you're returning the syntax of the symbol just to try to find the symbol again, just to get the symbols locations...

you have the symbol, and it's location to begin with, why are you round-tripping through everything multiple times?